api/v1alpha4/clusterclass_webhooks_test.go

```diff
@@ -0,0 +1,499 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utildefaulting "sigs.k8s.io/cluster-api/util/defaulting"
+)
+
+func TestClusterClassDefaultNamespaces(t *testing.T) {
+	namespace := "default"
+	ref := corev1.ObjectReference{
```

Comment:
nit: I think you can also move up/deduplicate the `LocalObjectTemplate{}` part.
Same below in the other test. But if you think it's too much effort and not worth it => just resolve it (I tend to agree)